src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -510,6 +510,17 @@ public IEnumerable<Document> GetDocumentsFrom(DocumentsOperationContext context,
             }
         }
 
+        public IEnumerable<Document> GetDocumentsFrom(DocumentsOperationContext context, List<string> collections, long etag, int start, int take)
+        {
+            foreach (var collection in collections)
+            {
+                foreach (var document in GetDocumentsFrom(context, collection, etag, start, take))
```

Comment:
remove the start, you cannot handle this properly without iterating half of the database and you will be setting 0 here always.

take needs to be handled in the inner loop. Currently you will take N x take where N - number of collections